waspc/data/Generator/templates/sdk/wasp/universal/url.ts

```diff
@@ -3,3 +3,7 @@ export function stripTrailingSlash(url: undefined): undefined
 export function stripTrailingSlash(url?: string): string | undefined {
     return url?.replace(/\/$/, "");
 }
+
+export function getOrigin(url: string) {
+    return new URL(url).origin;
+}
```

Comment:
A couple of comments related to @infomiho's [original implementation](https://github.com/wasp-lang/wasp/issues/1591#issuecomment-3397308595).
1. It puts thi slogic in `globalMiddleware`. I thought it made more sense to put it here, at the source
2. It does a `try..catch` to cover the case when the URLs aren't valid. I don't think this is necessary since we do that check during the env validation phase. @infomiho tell me if I missed something.

Possible improvements: